src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1368,20 +1368,54 @@ private void VisitStatements(ImmutableArray<IOperation> statements)
         {
             for (int i = 0; i < statements.Length; i++)
             {
-                if (statements[i] is VariableDeclarationGroupOperation declarationGroup
-                    && declarationGroup.DeclarationKind != VariableDeclarationKind.Default)
+                VisitPossibleUsingDeclaration(statements[i], statements, i, out var visitedUsingDeclaration);
+                if (visitedUsingDeclaration)
                 {
-                    // visit the using decl with the following statements
-                    var followingStatements = ImmutableArray.Create(statements, i + 1, statements.Length - i - 1);
-                    VisitUsingVariableDeclarationOperation(declarationGroup, followingStatements);
                     break;
                 }
-                else
-                {
-                    VisitStatement(statements[i]);
-                }
             }
+        }
 
+        /// <summary>
+        /// Visits a node that is possibly a using <see cref="IVariableDeclarationGroupOperation"/>
+        /// </summary>
+        /// <param name="operation">The statement to visit</param>
+        /// <param name="statements">All statements in the block containing this node</param>
+        /// <param name="startIndex">The current statement being visited in <paramref name="statements"/></param>
+        /// <param name="visitedUsingDeclaration">Set to true if this visited a using <see cref="IVariableDeclarationGroupOperation"/> operation</param>
+        /// <remarks>
+        /// The operation being visited is not necessarily equal to statements[startIndex]. 
+        /// When traversing down a set of labels, we set operation to the label.Operation and recurse, but statements[startIndex] still refers to the original parent label 
+        /// as we haven't actually moved down the original statement list
+        /// </remarks>
+        private void VisitPossibleUsingDeclaration(IOperation operation, ImmutableArray<IOperation> statements, int startIndex, out bool visitedUsingDeclaration)
+        {
+            switch (operation.Kind)
+            {
+                case OperationKind.VariableDeclarationGroup:
+                    var declarationGroup = (IVariableDeclarationGroupOperation)operation;
+                    if (declarationGroup.DeclarationKind != VariableDeclarationKind.Using && declarationGroup.DeclarationKind != VariableDeclarationKind.AsynchronousUsing)
+                    {
+                        goto default;
+                    }
+                    var followingStatements = ImmutableArray.Create(statements, startIndex + 1, statements.Length - startIndex - 1);
```

Comment:
This is a bit of a change to existing code, but is there a reason we can't just pass the start index and the original statement array to `VisitUsingVariableDeclarationOperation`, instead of allocating a new array for a smaller range?

Reply:
I looked at doing that, but we immediately form a block around those statements, so would just need to do the same thing inside VisitUsingDeclaration. It seemed simpler this way as it keeps the method signitures a bit simpler, but happy to move it down if you think it'd be better?

Reply:
No, that reason seems good.